module/zcommon/zfs_prop.c

```diff
@@ -245,6 +245,14 @@ zfs_prop_init(void)
 		{ NULL }
 	};
 
+	static zprop_index_t volmode_table[] = {
+		{ "default",	ZFS_VOLMODE_DEFAULT },
+		{ "geom",	ZFS_VOLMODE_GEOM },
```

Comment:
My concern is that `geom` as a name won't make sense to Linux users who aren't already familiar with FreeBSD. It would be nice to have alternate platform independent names.  How about something as simple as `geom` -> `full`, and we preserve the original name for compatibility.

This way the man page could simply mention that `geom` is an alias for `full`.  And `full` is more consistent with how the behavior is described in the man page.

> exposes volumes as **fully** fledged block devices, providing maximal functionality.

```c
	static zprop_index_t volmode_table[] = {
		{ "default",	ZFS_VOLMODE_DEFAULT },
		{ "full",	ZFS_VOLMODE_GEOM },
		{ "geom",	ZFS_VOLMODE_GEOM },
		{ "dev",	ZFS_VOLMODE_DEV },
		{ "none",	ZFS_VOLMODE_NONE },
		{ NULL }
	};
```